File: AdventOfCode2021/src/day01.rs
// day 1 of advent of code 2021
// author: rachael judy
// date: 1 dec 2021
// count occurrences of increases

use std::vec::Vec;

pub fn calculate(numbers : Vec<i32>, stage : i32) -> std::io::Result<()> {
    // set stage to 1 or 2
    let mut diff=1;
    if stage==2 {
        diff=3;
    }

    let mut count = 0;
    for i in diff..2000 {
        if numbers[i] > numbers[i-diff] {
            count+=1;
        }
    }

    println!("{}", count);

    Ok(())
}

File: AdventOfCode2022/day01.py
# Advent of Code 2022 Day 1
# Author:   Rachael Judy
# Date:     12/1/22
# Purpose:  Split a data file twice and find the top 3 sums of each batch

import parseMod

ready = False
day = 1
stage = 'a'

year = 2022

parseMod.createDataFile(year=year, day=day)
array = parseMod.readCSV_batch("data/" + str(day).zfill(2) + "data.csv")
calories = [[int(x) for x in arr.rstrip().split(' ')] for arr in array]

maxCals = []
for elf in calories:
    maxCals.append(sum(elf))

maxCals.sort()
if stage == 'a':
    result = maxCals[-1]
else:
    result = maxCals[-1] + maxCals[-2] + maxCals[-3]

if not ready:
    print(f'result: \n{result}')
elif ready:
    print("SUBMITTING RESULT: ", result)
    parseMod.submit(result, part=stage, day=day, year=year)
